Tests/Unit/InstanceTests.swift

```diff
@@ -80,4 +80,162 @@ class InstanceTests: XCTestCase {
     let isDifferent = dcOne !== dcTwo
     XCTAssertTrue(isDifferent)
   }
+
+  struct VariableTypeOne: OperationVariable {
+    var email: String
+    var handle: String
+    var someNumber: Int
+  }
+
+  // same query name, same variables, should return same instance
+  func testSameQueryRefInstance() throws {
+    let dc = DataConnect.dataConnect(connectorConfig: fakeConnectorConfigOne)
+
+    let queryName = "someQuery"
+    let variableOne = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+    let variableTwo = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+
+    let refOne = dc.query(
+      name: queryName,
+      variables: variableOne,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let refTwo = dc.query(
+      name: queryName,
+      variables: variableTwo,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let isSameInstance = refOne === refTwo
```

Comment:
nit: Consider renaming `refOne` and `refTwo` to `ref1` and `ref2`, respectively. IMO this improves readability.

Reply:
I'll keep this mind for the next time. Thanks!